runtime/pkg/container/container.go

```diff
@@ -0,0 +1,118 @@
+package container
+
+import (
+	"container/list"
+	"fmt"
+)
+
+// Container is a collection of items
+// More functions like remove can be added if future use cases arise
+type Container[T any] interface {
+	// Add consumes an item
+	Add(item T) bool
+
+	// IsFull return true if container is full
+	IsFull() bool
+
+	// returns items as slice
+	Items() []T
+}
+
+// boundedContainer is a container with limited capacity
+// Stops consuming more elements once capacity is reached
+// This is thread unsafe
+type boundedContainer[T any] struct {
+	items []T
+	index int
+}
+
+func NewBoundedContainer[T any](capacity int) (Container[T], error) {
+	if capacity <= 0 {
+		return nil, fmt.Errorf("len should be greater than 0")
+	}
+	b := &boundedContainer[T]{items: make([]T, capacity), index: 0}
```

Comment:
I ran a benchmark for preallocation vs append. I did not find much difference in execution time in both cases. Allocs/op were expectedly high in the append case. mem/op again depends on the input. 
In cases where capacity is high and we are using full capacity, append uses more memory since it pre-allocates memory. 
In cases where limited capacity is used, append uses less memory.
In all cases execution time was similar. I will use append itself for now.